File: CS/WpfPivotTestExample/FieldTemplateSelector.cs
using DevExpress.Xpf.PivotGrid;
using System.Windows;
using System.Windows.Controls;

namespace PivotGridViewModelBinding {
    public class FieldTemplateSelector : DataTemplateSelector {
        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            Field field = (Field)item;
            if(field.Interval == FieldGroupInterval.DateMonth || field.Interval == FieldGroupInterval.DateYear) {
                return (DataTemplate)((Control)container).FindResource("IntervalFieldTemplate");
            }
            return (DataTemplate)((Control)container).FindResource("DefaultFieldTemplate");
        }
    }
}

File: VB/WpfPivotTestExample/FieldTemplateSelector.vb
Imports DevExpress.Xpf.PivotGrid
Imports System.Windows
Imports System.Windows.Controls

Namespace PivotGridViewModelBinding
    Public Class FieldTemplateSelector
        Inherits DataTemplateSelector

        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
            Dim field As Field = DirectCast(item, Field)
            If field.Interval = FieldGroupInterval.DateMonth OrElse field.Interval = FieldGroupInterval.DateYear Then
                Return DirectCast(CType(container, Control).FindResource("IntervalFieldTemplate"), DataTemplate)
            End If
            Return DirectCast(CType(container, Control).FindResource("DefaultFieldTemplate"), DataTemplate)
        End Function
    End Class
End Namespace
